flightlib/include/flightlib/sensors/event_camera.hpp

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include <yaml-cpp/yaml.h>
+
+#include <chrono>
+#include <deque>
+#include <functional>
+#include <memory>
+#include <mutex>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+
+#include "flightlib/common/logger.hpp"
+#include "flightlib/common/types.hpp"
+#include "flightlib/sensors/sensor_base.hpp"
+
+namespace flightlib {
+
+struct Event {
+  int coord_x;
+  int coord_y;
+  int polarity;
+  float time;
+};
+
+class EventCamera : SensorBase {
+ public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+  EventCamera();
+  ~EventCamera();
+
+  // public set functions
+  bool setRelPose(const Ref<Vector<3>> B_r_BC, const Ref<Matrix<3, 3>> R_BC);
+  bool setWidth(const int width);
+  bool setHeight(const int height);
+  bool setFOV(const Scalar fov);
+  bool setCp(const float cp);
+  bool setCm(const float cm);
+  bool setsigmaCp(const float sigma_cp);
+  bool setsigmaCm(const float sigma_cm);
+  bool setRefractory(const uint64_t refractory_period);
+  bool setLogEps(const float log_eps);
+  bool setImgStore(const bool img_store);
+
+  bool changeTime(TimeMessage_t time_msg);
+  double getSecSimTime();
```

Comment:
also why using 'double" here

Reply:
because we need a higher resolution for this variable